seamless-immutable.development.js

```diff
@@ -326,20 +326,24 @@
     return result;
   }
 
+  // Creates empty object, using prototype if present
+  function instantiateEmptyObject() {
+    var prototype = Object.getPrototypeOf(this);
+    return prototype !== Object.prototype ? Object.create(prototype) : {};
+  }
+
   // Finalizes an object with immutable methods, freezes it, and returns it.
   function makeImmutableObject(obj) {
     addPropertyTo(obj, "merge", merge);
     addPropertyTo(obj, "without", without);
     addPropertyTo(obj, "asMutable", asMutableObject);
+    addPropertyTo(obj, "instantiateEmptyObject", instantiateEmptyObject);
```

Comment:
I want to point out that with the suggested approach, you'd still be doing the prototype check almost every time we instantiate an object. ```merge``` and ```without``` both call ```makeImmutableObject``` on their clones, and would go through this check. The added benefit over the previous approach is that ```asMutable``` and ```merge```s without ```anyChanges``` would not have to go through this check.

One option I was thinking about was passing an options hash to ```makeImmutableObject``` with ```instantiateEmptyObject``` as an option, which when present, would be used instead of constructing it.

Anyways, let me know if you still want to proceed with the above suggestions.